src/helpers/git/GitRepository.h

```diff
@@ -37,20 +37,32 @@ namespace Genio::Git {
 
 	class GitException {
```

Comment:
Does it make sense to make it a BException subclass?

Reply:
Yes, an overhaul of how exceptions are managed is in my to-do list. Special attention should be put into how TaskResult transport exceptions to the calling thread and deserverves a separate PR.